src/dune_2_excel.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+from typing import List, Any
+from datetime import datetime
+
+import click
+from dotenv import load_dotenv
+from dune_client.client import DuneClient
+from dune_client.models import ResultsResponse
+from dune_client.query import Query
+from dune_client.types import QueryParameter
+import pandas as pd
+from tqdm import tqdm
+
+
+@click.command()
+@click.option("--query", "-q", multiple=True)
+@click.option("--start-date", "-s", type=click.DateTime(formats=["%Y-%m-%d"]))
+def main(query: List[str], start_date: datetime) -> Any:
+    """
+    Main function of the script
+    Args:
+        query: List of queries to fetch
+        start_date: start_date query parameter
+
+    Returns:
+
+    """
+    monthly_reporting(query, start_date)
+    return 0
+
+
+def store_results(results: List[ResultsResponse], start_date: datetime) -> None:
+    """
+    Store results into xlsx file
+    Args:
+        results: results to be stored
+        start_date: start_date query parameter
+    """
+    writer = pd.ExcelWriter(  # pylint: disable=abstract-class-instantiated
+        f'out/{"_".join([str(x.query_id) for x in results])}'
+        f'_{start_date.strftime("%Y-%m-%d")}.xlsx',
+        engine="xlsxwriter",
+    )
+    for result in results:
+        pd.DataFrame(pd.DataFrame(result.result.rows)).to_excel(
+            writer, sheet_name=f"{result.query_id}", index=False
+        )
+    writer.close()
+
+
+def fetch_results(queries: List[str], start_date: datetime) -> List[ResultsResponse]:
+    """
+    Fetches results from Dune
+    Args:
+        queries: List of queries to fetch
+        start_date: start_date query parameter
+
+    Returns:
+        List[ResultsResponse]
+    """
+    dune_client: DuneClient = DuneClient(os.environ["DUNE_API_KEY"])
+    results: List[ResultsResponse] = []
+    for q in tqdm(queries):
```

Comment:
Interesting, I see [this is a progress bar](https://tqdm.github.io/) library! Kewl

Reply:
Also the fetching of multiple queries can be made asynchronous. Maybe just add a todo and an issue to capture that. Someone is working on making the client async right now! Not sure how well that will work with the progress bar though, but it will be faster!